src/OctoshiftCLI.Tests/ado2gh/Commands/ConfigureAutoLinkCommandTests.cs

```diff
@@ -30,15 +32,72 @@ public async Task Happy_Path()
             var githubRepo = "foo-repo";
             var adoOrg = "foo-ado-org";
             var adoTeamProject = "foo-ado-tp";
+            var keyPrefix = "AB#";
+            var urlTemplate = $"https://dev.azure.com/{adoOrg}/{adoTeamProject}/_workitems/edit/<num>/".Replace(" ", "%20");
 
             var mockGithub = new Mock<GithubApi>(null, null);
+            mockGithub.Setup(x => x.GetAutoLinks(It.IsAny<string>(), It.IsAny<string>()))
+                      .Returns(Task.FromResult(new List<AutoLink>()));
             var mockGithubApiFactory = new Mock<GithubApiFactory>(null, null, null);
             mockGithubApiFactory.Setup(m => m.Create()).Returns(mockGithub.Object);
 
             var command = new ConfigureAutoLinkCommand(new Mock<OctoLogger>().Object, mockGithubApiFactory.Object);
             await command.Invoke(githubOrg, githubRepo, adoOrg, adoTeamProject);
 
-            mockGithub.Verify(x => x.AddAutoLink(githubOrg, githubRepo, adoOrg, adoTeamProject));
+            mockGithub.Verify(x => x.DeleteAutoLink(githubOrg, githubRepo, 1), Times.Never);
+            mockGithub.Verify(x => x.AddAutoLink(githubOrg, githubRepo, keyPrefix, urlTemplate));
+        }
+
+        [Fact]
+        public async Task Idempotency_AutoLink_Exists()
+        {
+            var githubOrg = "foo-org";
+            var githubRepo = "foo-repo";
+            var adoOrg = "foo-ado-org";
+            var adoTeamProject = "foo-ado-tp";
+            var keyPrefix = "AB#";
+            var urlTemplate = $"https://dev.azure.com/{adoOrg}/{adoTeamProject}/_workitems/edit/<num>/".Replace(" ", "%20");
+
+            var mockGithub = new Mock<GithubApi>(null, null);
+            mockGithub.Setup(x => x.GetAutoLinks(It.IsAny<string>(), It.IsAny<string>()))
+                      .Returns(Task.FromResult(new List<AutoLink>
+                      {
+                          new AutoLink{Id = 1, KeyPrefix = keyPrefix, UrlTemplate = urlTemplate},
+                      }));
+            var mockGithubApiFactory = new Mock<GithubApiFactory>(null, null, null);
+            mockGithubApiFactory.Setup(m => m.Create()).Returns(mockGithub.Object);
+
+            var command = new ConfigureAutoLinkCommand(new Mock<OctoLogger>().Object, mockGithubApiFactory.Object);
+            await command.Invoke(githubOrg, githubRepo, adoOrg, adoTeamProject);
+
+            mockGithub.Verify(x => x.DeleteAutoLink(githubOrg, githubRepo, 1), Times.Never);
+            mockGithub.Verify(x => x.AddAutoLink(githubOrg, githubRepo, keyPrefix, urlTemplate), Times.Never);
```

Comment:
I think it'd be better to also test the log output since it's part of our user's experience. It's actually very easy by using [moq's Callback's feature](https://github.com/Moq/moq4/wiki/Quickstart#callbacks). I used callbacks in [this PR](https://github.com/github/gh-gei/pull/245) (hasn't been merged yet). In the PR take a look at [WaitForMigrationTests](https://github.com/github/gh-gei/pull/245/files#diff-b5d9e643ebdb145b018da0c51479707317c486c5a1639268cd96e61467964b79).